docs/source/configuration.rst

```diff
@@ -241,6 +249,16 @@ headers
 The required headers to be added to the request. Needs to be a dictionary
 
 
+timeout
+=======
+
+You can tell qrest to stop waiting for a response after a given number of
+seconds with the timeout parameter. When this time has passed, qrest will raise
+a ``requests.Timeout`` exception. The value of ``timeout`` should be
+``None`` (default), a non-negative float or int, or a tuple of size 2
+(indicating differing connection timeout and read timeout values (both tuple
+values should again be either ``None`` or a non-negative float or int)).
```

Comment:
I like this solution. I would limit the user to using a tuple of integers. The issue that arises is where to convert our custom timeout config to the one accepted by the requests library (0 should be converted to None, milliseconds to seconds). I think that the resourceConfig objects should reflect the user input, to prevent confusion, which leaves us with converting right before we make the call, in the _get routine in resource.py?

Reply:
The only 'downside' is that when a user makes multiple calls to the same endpoint, we are converting the timeout values every time. But this seems to be a trivial overhead price to pay to keep the ResourceConfig objects clean wrt the timeout parameter.

Reply:
Indeed, that overhead should be negligible.

Reply:
Done. The only question I still have is how we can test this. I've looked through both the requests request and response objects, but can't seem to find the used timeout value in there. Any idea on how we can check if the request was actually made with `timeout=(3.0, 3.0)` for a resourceConfig with `timeout=(3000, 3000)`?

Reply:
I missed this comment, but my initial response would be to mock `requests.requests`, similar to how it is done in `TestJsonPlaceholder`.